pybossa/cache/__init__.py

```diff
@@ -261,3 +353,26 @@ def delete_memoized_essential(function, *args, **kwargs):
             return False
         return bool(sentinel.master.delete(*keys_to_delete))
     return True
+
+
+def delete_memoize_with_l2_cache(function, *args, **kwargs):
+    """
+    Delete a memoize_with_l2_cache value from the cache.
+
+    Returns True if success or no cache is enabled
+
+    """
+    if os.environ.get('PYBOSSA_REDIS_CACHE_DISABLED') is None:
+        key = "%s:%s_args:" % (REDIS_KEYPREFIX, function.__name__)
+        if args or kwargs:
+            key_to_hash = get_key_to_hash(*args, **kwargs)
+            key = get_hash_key(key, key_to_hash)
+            key_l2 = f"{key}:l2"
+            key_deleted = bool(sentinel.master.delete(key))
+            key_l2_deleted = bool(sentinel.master.delete(key_l2))
+            return key_deleted and key_l2_deleted
+        keys_to_delete = list(sentinel.slave.scan_iter(match=key + '*', count=10000))
+        if not keys_to_delete:
+            return False
+        return bool(sentinel.master.delete(*keys_to_delete))
```

Comment:
It is not clear to me the expected behavior here. Let's discuss this tomorrow when you're available. Thanks!

Reply:
This is similar to [delete_memoized](https://github.com/bloomberg/pybossa/blob/b33be7ee95bc3047d9fe3ac5aaa50eab822db870/pybossa/cache/__init__.py#L320) to do a cache cleanup. ```delete_memoize_with_l2_cache``` does a cleanup for level 2 cache as well.

Reply:
This looks good. Thanks!